### ISSUE TITLE
Suggested changes for installing Netdiffusion dependencies.

### ISSUE THREAD

**Issue by CiroGuida** [noise-lab/NetDiffusion_Generator#5]
Please consider making the YAML configuration file public to facilitate the installation of Netdiffusion. Currently, the _requirements.txt_ file is not usable by the pip utility. 

In my case, I had to create a YML configuration file from the requirements file thanks to the script provided here: [https://gist.github.com/mfansler/5a1a703d1ac6bb2139547003248c3827](url) 
In your case, you can create the correct YAML with this command: 
`conda env export | head -n -1 | tail -n +2 > environment.yml`

Then I create a new virtual environment with the following: 
`conda create --name my_env pip python=3.10.9`

and activate the environment with
`conda activate my_env`

Then I installed tensorrt-libs from the pypi nvidia repository, as this package is not available in pypi and conda cannot resolve it:
`pip install --extra-index-url https://pypi.nvidia.com tensorrt-libs`
This step is mandatory to install all requirements listed in the YAML config file.

Next, I downgraded _torch_ and _triton_ to 2.0.1 and 2.0.0 respectively, because some packages were not compatible with torch 2.2.0 stated in the original requirements.txt

Finally, I updated all dependencies with:
`conda env update -n my_env -f environment.yml`

**Comment by Bozolzr** [comment]
thanks!

**Comment by TracyTd** [comment]
Thanks！this really helped me a lot！Also do you know where the full data set is? Thanks a lot!

**Comment by CiroGuida** [comment]
> Thanks！this really helped me a lot！Also do you know where the full data set is? Thanks a lot!

@TracyTd Unfortunately, the dataset used for the paper is private. The authors only shared an example of PCAP to play with the model. Try to collect your own traffic data or download PCAP datasets from other papers.

**Comment by TracyTd** [comment]
@CiroGuida  Thanks a lot!

**Comment by TracyTd** [comment]
@CiroGuida  May I ask you one more question? In the Data Preprocessing,is fine_tune/sd-webui-fork/stable-diffusion-webui/**webui.sh**  downloading from https://github.com/AUTOMATIC1111/stable-diffusion-webui or other places? I can't find correct file in https://github.com/Chasexj/stable-diffusion-webui-fork/tree/2533cf84cad7313b8653717d15728a2d9a862c15. Thank you so much!

**Comment by CiroGuida** [comment]
> @CiroGuida May I ask you one more question? In the Data Preprocessing,is fine_tune/sd-webui-fork/stable-diffusion-webui/**webui.sh** downloading from https://github.com/AUTOMATIC1111/stable-diffusion-webui or other places? I can't find the correct file at https://github.com/Chasexj/stable-diffusion-webui-fork/tree/2533cf84cad7313b8653717d15728a2d9a862c15. Thank you so much!

Yes, the link is automatic1111. However, the correct version of stable diffusion is already forked into the repository. So, you have to manage two env variables in webui-user.sh (.bat if you are using Windows and modify it according to Windows). Modify them as follows:
`install_dir=$(dirname "$PWD")

# Name of the subdirectory
clone_dir="stable-diffusion-webui-fork"`

**Comment by TracyTd** [comment]
Thank you very much!
